heksher/setting.py

```diff
@@ -81,58 +91,39 @@ def get(self, **contexts) -> T:
             return self.default_value
         return from_rules
 
-    def update(self, client, context_features: Sequence[str], root: RuleBranch[T]):
+    def update(self, client, context_features: Sequence[str], rules: Iterable[Tuple[Sequence[Tuple[str, str]], T]],
+               is_rule_collection: bool = True):
         """
         Update a setting's rules from a client
 
         Args:
             client: The client updating the setting.
             context_features: The context features the root was collated by.
-            root: A collated rulebranch root.
+            rules: An iterable of rules to collate.
+            is_rule_collection: bool, if rules are a collection; if not, don't collate the rules
```

Comment:
why is this value needed?

Reply:
when the rules param is not a collection (so you won't check it twice)